### PR TITLE
Add typing annotations to Python scripts

### PR DESCRIPTION
**Pull request by wismill**

### CODE REVIEW

File: scripts/update-unicode.py [hunk 1]
```diff
@@ -1228,7 +1228,7 @@ def stats(self, int_size) -> Stats:
             offsets2_int_size=0,
         )
 
-    @classmethod
+    @staticmethod
     def test(cls):
```
Comment (whot) [inline scripts/update-unicode.py:1232]: huh, I thought `staticmethod` doesn't take a `cls` argument, does it? If there's any callers they should now fail with "too few arguments"?

File: scripts/perfect_hash.py [hunk 2]
```diff
@@ -384,7 +391,12 @@ def __call__(self, data, quote=False):
         return "\n".join(l.rstrip() for l in aux.getvalue().split("\n"))
 
 
-def generate_code(keys, Hash=StrSaltHash, template=None, options=None):
+def generate_code(
+    keys: list[str],
+    Hash: type = StrSaltHash,
```
Comment (whot) [inline scripts/perfect_hash.py:396]: sigh, github doesn't let me do suggestions here:
shoudl this be 

```
Hash: type[H] = StrSaltHash,
```
for consistency with `generate_hash()`

**Review by whot — COMMENTED** [review]
belated :+1: but the `staticmethod` bit looks wrong

File: scripts/update-unicode.py (hunk 1 above)
Comment (wismill) [reply scripts/update-unicode.py:1232]: You are right. Fixed in #534

File: scripts/perfect_hash.py (hunk 2 above)
Comment (wismill) [reply scripts/perfect_hash.py:396]: Fixed in #534

**Pull request merged**